lib/remote/apilistener.cpp

```diff
@@ -529,7 +540,34 @@ void ApiListener::NewClientHandlerInternal(boost::asio::yield_context yc, const
 
 	boost::system::error_code ec;
 
-	sslConn.async_handshake(role == RoleClient ? sslConn.client : sslConn.server, yc[ec]);
+	{
+		struct DoneHandshake
+		{
+			bool Done = false;
+		};
+
+		auto doneHandshake (Shared<DoneHandshake>::Make());
+
+		IoEngine::SpawnCoroutine(*strand, [strand, client, doneHandshake](asio::yield_context yc) {
+			namespace sys = boost::system;
+
+			{
+				boost::asio::deadline_timer timer (strand->context());
+				timer.expires_from_now(boost::posix_time::seconds(10));
```

Comment:
Can we use the configureable TLS timeout here instead of a hard coded one? e.g. `ApiListener::GetTlsHandshakeTimeout()` ?

Reply:
What was the pre-v2.11 behavior?

Reply:
The configurable timeout was used:

https://github.com/Icinga/icinga2/blob/24e5986c8d501b37d8c39b663cefd955d6ca2085/lib/base/tlsstream.cpp#L316